src/utils/IO/system_checks.jl

```diff
@@ -61,3 +61,19 @@ function critical_components_check(sys::System)
     missing_critical_components &&
     throw(IS.InvalidValue("Critical Componeents are not present."))
 end
+
+"""
+Checks the system for sum(generator ratings) >= sum(load ratings).
+
+# Arguments
+- `sys::System`: system
+"""
+function adequacy_check(sys::System)
+    total_gen = sum(get_rating.(get_components(Generator, sys))) * get_basepower(sys)
+    @debug "Total System Generation: $total_gen"
+    total_load =
+        sum(get_maxactivepower.(get_components(ElectricLoad, sys))) * get_basepower(sys)
+    @debug "Total System Load: $total_load"
+    total_load > total_gen &&
+    @error "System peak load ($total_load) exceeds total generation capability ($total_gen)."
```

Comment:
Should this throw an exception instead?

Reply:
@daniel-thom and @jd-lara, I wasn't sure what to do here. For the example that @jd-lara mentioned, I didn't throw an exception. However, I think the severity is such that a warning is maybe insufficient...

Reply:
I think that there are too many cases where it does make sense to have more load than generators. I strongly suggest we keep it a warning since there aren't strong warnings.

Reply:
Regardless of warning vs error, we will just confuse people with messages if their data is actually correct.  Is it possible (and reasonably feasible) to determine when this acceptable?